drivers/bebop_bus/test/test.cpp

```diff
@@ -0,0 +1,329 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2016 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+#include <unistd.h>
+#include <string.h>
+#include "DriverFramework.hpp"
+#include "BebopBus.hpp"
+
+
+using namespace DriverFramework;
+
+class BebopBusTester : public BebopBus
+{
+public:
+	static const int TEST_PASS = 0;
+	static const int TEST_FAIL = 1;
+
+	BebopBusTester() :
+		BebopBus(BEBOP_BUS_DEVICE_PATH)
+	{}
+
+	int run(void);
+
+private:
+
+	int		m_pass;
+	bool	m_done = false;
+
+	int print_observation();
+	int test_gpios();
+	int test_sound();
+	int test_motors();
+	char *strstatus(uint8_t status);
+};
+
+char *BebopBusTester::strstatus(uint8_t status)
+{
+	static char text[15];
+
+	switch (status) {
+	case INIT:
+		strcpy(text, "Initialize");
```

Comment:
@julianoes no that does not work, because the string goes out of scope after returning. Using `static` avoids that. However it should be noted that the `static` version is not thread-safe... Might not be relevant at this place.